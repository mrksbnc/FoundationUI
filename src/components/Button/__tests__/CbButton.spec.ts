import { mount } from '@vue/test-utils';
import { describe, expect, test } from 'vitest';
import { ButtonSize, ButtonType, ClButton } from '..';

describe('ClButton', () => {
	test('renders correctly', () => {
		const wrapper = mount(ClButton);
		expect(wrapper).toMatchSnapshot();
	});

	test('renders correctly with color', () => {
		const wrapper = mount(ClButton, {
			props: {
				label: 'Test',
				color: ButtonType.danger,
			},
		});
		expect(wrapper).toMatchSnapshot(wrapper.html());
	});

	test('renders correctly with size', () => {
		const wrapper = mount(ClButton, {
			props: {
				label: 'Test',
				size: ButtonSize.large,
			},
		});
		expect(wrapper).toMatchSnapshot(wrapper.html());
	});
});
