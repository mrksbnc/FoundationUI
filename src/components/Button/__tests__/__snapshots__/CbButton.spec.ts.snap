// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DsButton > renders correctly 1`] = `
"<button id=\\"dsbutton_1692480764523\\" class=\\"ds-button w-full h-full flex flex-row justify-center items-center focus:ring-1 font-medium focus:outline-none px-3 py-2 text-sm font-medium text-center rounded-lg text-white bg-blue-700 dark:bg-blue-600 hover:bg-blue-800 dark:hover:bg-blue-700 focus:ring-blue-300 dark:focus:ring-blue-800\\">
  <div class=\\"flex flex-row justify-center items-center\\"><span class=\\"ds-button__prefix mr-2\\"></span><span>Test</span><span class=\\"ds-button__suffix ml-2\\"></span></div>
</button>"
`;
